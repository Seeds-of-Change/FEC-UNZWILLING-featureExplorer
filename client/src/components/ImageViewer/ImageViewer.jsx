import React from 'react';
import PropTypes from 'prop-types';
import style from './styles.css';

export default function ImageViewer(props) {
  const {
    features, activeIndex, updateActive, productImage,
  } = props;

  const handleFeatureChange = (e) => {
    e.preventDefault();
    const newActiveIndex = Number(e.target.attributes.index.value);

    updateActive(newActiveIndex);
  };

  return (
    <div className={style.wrapper}>
      <img className={style.image} alt="main product" src={productImage} />
      {features.map((feature, index) => (
        <button
          className={activeIndex === index ? `${style.button} ${style.active}` : `${style.button}`}
          type="button"
          key={`${feature.posX}${feature.posY}`}
          index={index}
          style={{ top: `${feature.posY}%`, left: `${feature.posX}%` }}
          onClick={handleFeatureChange}
        >
          +
        </button>
      ))}
    </div>
  );
}

ImageViewer.defaultProps = {
  features: null,
  updateActive: null,
  activeIndex: 0,
  productImage: undefined,
};

ImageViewer.propTypes = {
  features: PropTypes.arrayOf(
    PropTypes.shape({
      heading: PropTypes.string,
      description: PropTypes.string,
      posX: PropTypes.number,
      posY: PropTypes.number,
    }),
  ),
  updateActive: PropTypes.func,
  activeIndex: PropTypes.number,
  productImage: PropTypes.string,
};
