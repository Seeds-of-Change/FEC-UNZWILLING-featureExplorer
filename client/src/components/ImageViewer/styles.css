.wrapper {
  width: 55%;
  display:inline-block;
  vertical-align: top;
  position: relative;
}

.image {
  width: 100%;
  border-style: none;
}

.button {
  position: absolute;
  width: 24px;
  height: 24px;
  text-align: center;
  background-color: #000;
  color: #fff;
  cursor: pointer;
  transition: all .3s ease;
}

.button.active{
  background-color: #e80009;
}