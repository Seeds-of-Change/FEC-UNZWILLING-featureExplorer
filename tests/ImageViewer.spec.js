import React from 'react';
import renderer from 'react-test-renderer';
import ImageViewer from '../client/src/components/ImageViewer/ImageViewer';

describe('ImageViewer', () => {
  describe('Have props', () => {
    let component;

    beforeEach(() => {
      const { features, activeIndex } = {
        features: [{
          heading: 'heading for 1', description: 'description string for feature 1', posX: 60, posY: 62,
        }, {
          heading: 'heading for 2', description: 'description string for feature 2', posX: 70, posY: 32,
        }, {
          heading: 'heading for 3', description: 'description string for feature 3', posX: 20, posY: 83,
        }],
        activeIndex: 0,
      };

      component = renderer.create(<ImageViewer features={features} activeIndex={activeIndex} />);
    });

    test('snapshot renders', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
