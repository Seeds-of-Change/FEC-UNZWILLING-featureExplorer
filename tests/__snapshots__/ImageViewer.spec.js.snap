// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageViewer Have props snapshot renders 1`] = `
<div
  className="wrapper"
>
  <img
    alt="main product"
    className="image"
    src="https://unzwilling.s3-us-west-1.amazonaws.com/1.jpg"
  />
  <button
    className="button active"
    index={0}
    onClick={[Function]}
    style={
      Object {
        "left": "60%",
        "top": "62%",
      }
    }
    type="button"
  >
    +
  </button>
  <button
    className="button"
    index={1}
    onClick={[Function]}
    style={
      Object {
        "left": "70%",
        "top": "32%",
      }
    }
    type="button"
  >
    +
  </button>
  <button
    className="button"
    index={2}
    onClick={[Function]}
    style={
      Object {
        "left": "20%",
        "top": "83%",
      }
    }
    type="button"
  >
    +
  </button>
</div>
`;
